examples/step-49/step-49.ipynb

```diff
@@ -0,0 +1,1128 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['generate_cheese',\n",
+       " 'generate_enclosed_hyper_cube',\n",
+       " 'generate_fixed_subdivided_parallelepiped',\n",
+       " 'generate_general_cell',\n",
+       " 'generate_half_hyper_ball',\n",
+       " 'generate_hyper_ball',\n",
+       " 'generate_hyper_cube',\n",
+       " 'generate_hyper_cube_with_cylindrical_hole',\n",
+       " 'generate_hyper_rectangle',\n",
+       " 'generate_hyper_shell',\n",
+       " 'generate_hyper_sphere',\n",
+       " 'generate_parallelepiped',\n",
+       " 'generate_parallelogram',\n",
+       " 'generate_quarter_hyper_ball',\n",
+       " 'generate_simplex',\n",
+       " 'generate_subdivided_hyper_cube',\n",
+       " 'generate_subdivided_hyper_rectangle',\n",
+       " 'generate_subdivided_material_hyper_rectangle',\n",
+       " 'generate_subdivided_steps_hyper_rectangle',\n",
+       " 'generate_varying_subdivided_parallelepiped']"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "list(filter(re.compile('generate_*').match, dir(dealii.Triangulation)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Importing from external programs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The original C++ class [GridIn](https://www.dealii.org/current/doxygen/deal.II/classGridIn.html) can read many different mesh formats from a file from disk. In PyDealII, GridIn functions are called via the *Triangulation.read* method"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Help on built-in function read:\n",
+      "\n",
+      "read(...)\n",
+      "    read( (Triangulation)self, (str)filename, (str)format) -> None :\n",
+      "        Read a mesh from the file according to the given data format.          \n",
+      "        The possible formats are:                                              \n",
+      "          - msh                                                                \n",
+      "          - vtk\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(dealii.Triangulation.read)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Meshes can be generated from different tools like [gmsh](http://gmsh.info/), [lagrit](https://lagrit.lanl.gov/) and [cubit](https://cubit.sandia.gov/). See the documentation of [GridIn](https://www.dealii.org/current/doxygen/deal.II/classGridIn.html) for more information. The problem is that deal.II needs meshes that only consist of quadrilaterals and hexahedra – tetrahedral meshes won't work (this means tools like tetgen can not be used directly).\n",
```

Comment:
I've never heard of lagrit before. Looks pretty nice. Did you know about this @jppelteret ?